source/common/common/utility.h

```diff
@@ -808,6 +964,60 @@ class InlineString : public InlineStorage {
   char data_[];
 };
 
+class TinyString {
+public:
+  TinyString() = default;
+  TinyString(absl::string_view str) { populateFromStringView(str); }
+  TinyString(TinyString&& src) : data_(std::move(src.data_)) { src.data_.reset(); }
+  TinyString(const TinyString& src) {
+    if (src.hasValue()) {
+      populateFromTinyString(src);
+    }
+  }
+
+  TinyString& operator=(const TinyString& src) {
+    if (&src != this) {
+      if (src.hasValue()) {
+        populateFromTinyString(src);
+      } else {
+        data_.reset();
+      }
+    }
+    return *this;
+  }
+
+  void operator=(TinyString&& src) {
+    if (&src != this) {
+      data_ = std::move(src.data_);
+    }
+  }
+
+  /**
+   * Caller must ensure hasValue() is true.
```

Comment:
Could this mention what would happen otherwise?
Related: should toStringView ASSERT nullptr?